imports/ui/pages/resources/components/BookingsField.jsx

```diff
@@ -0,0 +1,331 @@
+    } catch (error) {
+      message.error(error.reason);
+    }
+  };
+    
+  const handleDateAndTimeChange = (value, key) => {
+    newBooking[key] = value;
+    setNewBooking({...newBooking});
+    validateBookings(newBooking, domain);
+  };
+
+  const validateBookings = (selectedOccurence, selectedResource) => {
+
+    const allOccurencesWithSelectedResource = occurences.filter(
+      (occurence) => {
+        if (selectedResource.isCombo) {
+          return selectedResource.resourcesForCombo.some((resourceForCombo) => {
+            return resourceForCombo.label === occurence.resource;
+          });
+        }
+        return occurence.resource === selectedResource.label;
+      }
+    );
+
+    // console.log('allOccurencesWithSelectedResource: ', allOccurencesWithSelectedResource);
+
+    const occurenceWithConflict = allOccurencesWithSelectedResource.find(
+      (occurence) => {
+        const selectedStart = `${selectedOccurence.startDate} ${selectedOccurence.startTime}`;
+        const selectedEnd = `${selectedOccurence.endDate} ${selectedOccurence.endTime}`;
+        const existingStart = `${occurence.startDate} ${occurence.startTime}`;
+        const existingEnd = `${occurence.endDate} ${occurence.endTime}`;
+        const dateTimeFormat = 'YYYY-MM-DD HH:mm';
+        return (
+          moment(selectedStart, dateTimeFormat).isBetween(
+            existingStart,
+            existingEnd
+          ) ||
+          moment(selectedEnd, dateTimeFormat).isBetween(
+            existingStart,
+            existingEnd
+          )
+        );
+      }
+    );
+
+    // console.log('occurenceWithConflict: ', occurenceWithConflict);
+
+    if (occurenceWithConflict) {
+      if (selectedOccurence?.conflict) delete selectedOccurence.conflict
+      setNewBooking({
+        ...selectedOccurence,
+        conflict: {
+          ...occurenceWithConflict,
+        },
+      });
+    } else {
+      delete selectedOccurence.conflict
+      setNewBooking(selectedOccurence);
+    }
+
+    // console.log(newBooking);
+  };
+
+  const onSubmit = async (values) => {
+    multipledays === false
+      ? values = { ...newBooking, endDate: newBooking.startDate, ...values }
+      : values = { ...newBooking, ...values };
+
+    activityValues = {
+      title: values.title,
+      subTitle: '',
+      longDescription: values.description,
+      resource: {
+        label: domain.label,
+        _id: domain._id,
+        resourceIndex: domain.resourceIndex,
+      },
+      place: '',
+      practicalInfo: '',
+      internalInfo: '',
+      address: '',
+      capacity: 0,
+      datesAndTimes: [{
+        startDate: values.startDate,
+        startTime: values.startTime,
+        endDate: values.endDate,
+        endTime: values.endTime,
+        capacity: 0,
+      }],
+      isPublicActivity: false,
+      isRegistrationDisabled: true,
+    };
+
+    Meteor.call('createActivity', activityValues, '', (error, result) => {
```

Comment:
Would be nice to use `call()` async function here. And then to add a success message to notify user

Reply:
👍 okey, I am still new with some usages, thanx